sbin/prepareWorkspace.sh

```diff
@@ -333,6 +333,42 @@ checkingAndDownloadingAlsa() {
   echo "${ALSA_BUILD_URL}" > "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}/metadata/dependency_version_alsa.txt"
 }
 
+checkingGCCforLinux() {
```

Comment:
So I think rather than assuming the gcc used from the environment variables, we should use what was determined in the actual build. We currently get the "summary" here: https://github.com/adoptium/temurin-build/blob/2af5238b07c38fea4e9cb60b256adc1b67b19afb/sbin/build.sh#L836

Use a similar method to get just the gcc and gc++ versions used, eg:
Here's the output in the configure log:
```
 Tools summary:
* Boot JDK:       openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8) OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (build 17.0.5+8, mixed mode, sharing) (at /usr/lib/jvm/jdk-17)
* Toolchain:      gcc (GNU Compiler Collection)
* C Compiler:     Version 10.3.0 (at /usr/local/gcc10/bin/gcc-10.3)
* C++ Compiler:   Version 10.3.0 (at /usr/local/gcc10/bin/g++-10.3)
```
eg. something like (i've not tested this!)
```
sed -n '/^Tools summary:$/,$p' "${inputConfigFile}" | grep "C Compiler:" | sed 's/^.*C Compiler:.*Version\s+([0-9\.]).*$/\1/'
```